### ISSUE TITLE
I couldn't find softplus and other things.

### ISSUE THREAD

**Issue by samuelcaldas** [SciSharp/TensorFlow.NET#540]
I couldn't find sofplus, normal sample() and prob(). if they have not yet been implemented, any guidance for me to try to implement? forgive my bad english.

**Comment by Oceania2018** [comment]
@samuelcaldas You can follow other implementation pattern. Dig into python source code and convert to C#.

**Comment by samuelcaldas** [comment]
> @samuelcaldas You can follow other implementation pattern. Dig into python source code and convert to C#.

I am following your tip and would like to send corrections. Is there a pattern to follow?

**Comment by Oceania2018** [comment]
@samuelcaldas I'm actively working on tf2.0, will not upgrade 0.15.x branch. If you're interested in tf2.0 implementation. We can work together.

**Comment by samuelcaldas** [comment]
it will be an honour. I must look at master branch, right?

**Comment by Oceania2018** [comment]
@samuelcaldas Correct. Let me know if you need any help to get started.

**Comment by samuelcaldas** [comment]
I'm sorry if this is a stupid question, but I couldn't run my compiled version of tf.net 2. I had a few problems with locally compiled tensorflow.dll and nuget.org version. Strangely, everything goes well with version 1.15.

**Comment by samuelcaldas** [comment]
@Oceania2018 I'm getting System.NullReferenceException with this code and I have no idea how to solve it. can you help me solve? tell me if you need the full log.
`static void Main()
  {
            var obs = tf.placeholder(tf.float32);
            var out = tf.layers.dense(inputs: obs, units: 20);
}`

**Comment by Oceania2018** [comment]
Full log would be helpful.

**Comment by samuelcaldas** [comment]
@Oceania2018 This is the output error (I was not able to generate in English, only Portuguese-Brazil - PT-br):
`2020-06-03 19:17:26.275019: I tensorflow/compiler/xla/service/service.cc:168] XLA service 0x21b158789d0 initialized for platform Host (this does not guarantee that XLA will be used). Devices:
2020-06-03 19:17:26.290842: I tensorflow/compiler/xla/service/service.cc:176]   StreamExecutor device (0): Host, Default Version

Exceção Sem Tratamento: System.NullReferenceException: Referência de objeto não definida para uma instância de um objeto.
   em Tensorflow.Operation.op_Implicit(Operation op)
   em Tensorflow.Tensor._as_tf_output()
   em Tensorflow.ops._create_c_op[T](Graph graph, NodeDef node_def, T[] inputs, Operation[] control_inputs)
   em Tensorflow.Operation..ctor(NodeDef node_def, Graph g, Tensor[] inputs, TF_DataType[] output_types, ITensorOrOperation[] control_inputs, TF_DataType[] input_types, String original_op, OpDef op_def)
   em Tensorflow.Graph.create_op(String op_type, Tensor[] inputs, TF_DataType[] dtypes, TF_DataType[] input_types, String name, Dictionary`2 attrs, OpDef op_def)
   em Tensorflow.OpDefLibrary.<>c__DisplayClass1_0.<_apply_op_helper>b__0(NameScope scope)
   em Tensorflow.Binding.tf_with[TIn,TOut](TIn py, Func`2 action)
   em Tensorflow.OpDefLibrary._apply_op_helper(String op_type_name, String name, Dictionary`2 keywords)
   em Tensorflow.OpDefLibrary._apply_op_helper(String op_type_name, String name, Object args)
   em Tensorflow.gen_random_ops.random_uniform(Tensor shape, TF_DataType dtype, Nullable`1 seed, Nullable`1 seed2, String name)
   em Tensorflow.random_ops.<>c__DisplayClass1_0.<random_uniform>b__0(NameScope scope)
   em Tensorflow.Binding.tf_with[TIn,TOut](TIn py, Func`2 action)
   em Tensorflow.random_ops.random_uniform(Int32[] shape, Single minval, Single maxval, TF_DataType dtype, Nullable`1 seed, String name)
   em Tensorflow.Operations.Initializers.VarianceScaling.call(TensorShape shape, TF_DataType dtype, Nullable`1 verify_shape)
   em Tensorflow._VariableStore.<>c__DisplayClass6_0.<_get_single_variable>b__0()
   em Tensorflow.RefVariable.<>c__DisplayClass35_0.<_init_from_args>b__2(NameScope scope2)
   em Tensorflow.Binding.tf_with[T](T py, Action`1 action)
   em Tensorflow.RefVariable.<>c__DisplayClass35_0.<_init_from_args>b__1(NameScope scope)
   em Tensorflow.Binding.tf_with[T](T py, Action`1 action)
   em Tensorflow.RefVariable.<>c__DisplayClass35_0.<_init_from_args>b__0(ITensorFlowObject <p0>)
   em Tensorflow.Binding.tf_with(ITensorFlowObject py, Action`1 action)
   em Tensorflow.RefVariable._init_from_args(Object initial_value, Boolean trainable, List`1 collections, Boolean validate_shape, String caching_device, String name, TF_DataType dtype)
   em Tensorflow.RefVariable..ctor(Object initial_value, Boolean trainable, List`1 collections, Boolean validate_shape, String caching_device, String name, VariableDef variable_def, TF_DataType dtype, String import_scope)
   em Tensorflow.variable_scope.default_variable_creator(Object initial_value, String name, Nullable`1 trainable, List`1 collections, TF_DataType dtype, Int32[] shape, Boolean validate_shape, Nullable`1 use_resource, VariableSynchronization synchronization, VariableAggregation aggregation)
   em Tensorflow._VariableStore._get_single_variable(String name, TensorShape shape, TF_DataType dtype, IInitializer initializer, Boolean reuse, Nullable`1 trainable, List`1 collections, Boolean validate_shape, Nullable`1 use_resource, VariableSynchronization synchronization, VariableAggregation aggregation)
   em Tensorflow._VariableStore._true_getter(String name, TensorShape shape, TF_DataType dtype, Object initializer, Nullable`1 trainable, List`1 collections, Boolean validate_shape, VariableSynchronization synchronization, VariableAggregation aggregation)
   em Tensorflow._VariableStore.get_variable(String name, TensorShape shape, TF_DataType dtype, Object initializer, Nullable`1 reuse, Nullable`1 trainable, List`1 collections, Boolean validate_shape, VariableSynchronization synchronization, VariableAggregation aggregation)
   em Tensorflow.VariableScope.<>c__DisplayClass17_0.<get_variable>b__0(NameScope scope)
   em Tensorflow.Binding.tf_with[TIn,TOut](TIn py, Func`2 action)
   em Tensorflow.VariableScope.get_variable(_VariableStore var_store, String name, TensorShape shape, TF_DataType dtype, Object initializer, Nullable`1 trainable, List`1 collections, Nullable`1 use_resource, Boolean validate_shape, VariableSynchronization synchronization, VariableAggregation aggregation)
   em Tensorflow.tensorflow.get_variable(String name, TensorShape shape, TF_DataType dtype, Object initializer, Nullable`1 trainable, List`1 collections, Nullable`1 use_resource, Boolean validate_shape, VariableSynchronization synchronization, VariableAggregation aggregation)
   em Tensorflow.Layers.Layer.<>c.<add_weight>b__10_2(String name1, Int32[] shape1, TF_DataType dtype1, IInitializer initializer1, Boolean trainable1)
   em Tensorflow.Train.Trackable._add_variable_with_custom_getter(String name, Int32[] shape, TF_DataType dtype, IInitializer initializer, Func`6 getter, Boolean overwrite, Boolean trainable, Boolean use_resource, VariableSynchronization synchronization, VariableAggregation aggregation)
   em Tensorflow.Keras.Layers.Layer.add_weight(String name, Int32[] shape, TF_DataType dtype, IInitializer initializer, Nullable`1 trainable, Func`6 getter)
   em Tensorflow.Layers.Layer.<>n__0(String name, Int32[] shape, TF_DataType dtype, IInitializer initializer, Nullable`1 trainable, Func`6 getter)
   em Tensorflow.Layers.Layer.<>c__DisplayClass10_0.<add_weight>b__1(NameScope <p0>)
   em Tensorflow.Binding.tf_with[TIn,TOut](TIn py, Func`2 action)
   em Tensorflow.Layers.Layer.<>c__DisplayClass10_0.<add_weight>b__0(variable_scope scope)
   em Tensorflow.Binding.tf_with[TIn,TOut](TIn py, Func`2 action)
   em Tensorflow.Layers.Layer.add_weight(String name, Int32[] shape, TF_DataType dtype, IInitializer initializer, Nullable`1 trainable, VariableSynchronization synchronization, VariableAggregation aggregation)
   em Tensorflow.Keras.Layers.Dense.build(TensorShape input_shape)
   em Tensorflow.Keras.Layers.Layer._maybe_build(Tensor input)
   em Tensorflow.Keras.Layers.Layer.<>c__DisplayClass23_0.<__call__>b__0(NameScope <p0>)
   em Tensorflow.Binding.tf_with[T](T py, Action`1 action)
   em Tensorflow.Keras.Layers.Layer.__call__(Tensor[] inputs, Tensor training, Tensor state, VariableScope scope)
   em Tensorflow.Layers.Layer.<>c__DisplayClass8_0.<__call__>b__0(variable_scope scope2)
   em Tensorflow.Binding.tf_with[T](T py, Action`1 action)
   em Tensorflow.Layers.Layer.__call__(Tensor inputs, Tensor training, Tensor state, VariableScope scope)
   em Tensorflow.Layers.Layer.apply(Tensor inputs, Tensor training)
   em Tensorflow.tensorflow.layers_internal.dense(Tensor inputs, Int32 units, IActivation activation, Boolean use_bias, IInitializer kernel_initializer, IInitializer bias_initializer, Boolean trainable, String name, Nullable`1 reuse)
   em PPO.NETv2.Program.Main(String[] args) na C:\Users\Samuel\source\repos\PPO.NETv2\PPO.NETv2\Program.cs:linha 31
`

**Comment by Oceania2018** [comment]
@samuelcaldas Check this [code sample.](https://github.com/SciSharp/TensorFlow.NET/blob/6e4bad40b6b36278efdbcb4d585051b549d09e14/test/TensorFlowNET.UnitTest/Keras/LayersTest.cs#L50)
